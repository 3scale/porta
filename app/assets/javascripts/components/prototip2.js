

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
