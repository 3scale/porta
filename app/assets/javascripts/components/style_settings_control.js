// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
