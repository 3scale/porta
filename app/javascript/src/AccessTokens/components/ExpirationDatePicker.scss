@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
