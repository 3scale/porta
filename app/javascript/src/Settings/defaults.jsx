const INTEGRATION_METHOD_DEFAULTS = {
  value: 'hosted',
  name: 'deployment_option',
  catalog: {
    hosted: 'APIcast 3scale managed',
    self_managed: 'APIcast self-managed',
    service_mesh_istio: 'Istio'
  }
}
const AUTHENTICATION_METHOD_DEFAULTS = {
  value: '1',
  name: 'proxy_authentication_method',
  catalog: {
    '1': 'API Key (user_key)',
    '2': 'App_ID and App_Key Pair',
    oidc: 'OpenID Connect'
  }
}

const PROXY_ENDPOINTS_DEFAULTS = [
  {
    defaultValue: 'https://api-2.staging.apicast.com',
    placeholder: 'https://api.provider-name.com',
    label: 'Staging Public Base URL',
    name: 'sandbox_endpoint',
    hint: 'Public address of your API gateway in the staging environment.',
    value: 'https://custom.api.staging.provider-name.com',
    inputType: 'url'
  },
  {
    defaultValue: 'https://api-2.apicast.com',
    placeholder: 'https://api.provider-name.com',
    label: 'Staging Public Base URL',
    name: 'endpoint',
    hint: 'Public address of your API gateway in the production environment.',
    value: 'https://custom.api.provider-name.com',
    inputType: 'url'
  }
]

const API_KEY_SETTINGS_DEFAULT = {
  label: 'Auth user key',
  name: 'auth_user_key',
  value: 'user_key'
}

const APP_ID_KEY_PAIR_SETTINGS_DEFAULT = [
  {
    label: 'App ID parameter',
    name: 'auth_app_id',
    hint: 'Public address of your API gateway in the staging environment.',
    value: 'app_id'
  },
  {
    label: 'App Key parameter',
    name: 'auth_app_key',
    value: 'app_key'
  }
]

const OIDC_BASICS_SETTINGS_DEFAULTS = {
  type: {
    value: 'keycloak',
    name: 'oidc_issuer_type',
    label: 'OpenID Connect Issuer Type',
    catalog: {
      keycloak: 'Red Hat Single Sign-On',
      rest: 'REST API'
    }
  },
  item: {
    value: '',
    name: 'oidc_issuer_endpoint',
    label: 'OpenID Connect Issuer',
    placeholder: 'https://sso.example.com/auth/realms/gateway',
    hint: 'Location of your OpenID Provider. The format of this endpoint is determined on your OpenID Provider setup. A common guidance would be "https://CLIENT_ID:CLIENT_SECRET@HOST:PORT/auth/realms/REALM_NAME".'
  }
}

const OIDC_FLOW_SETTINGS_DEFAULTS = [
  { name: 'service_accounts_enabled', label: 'Service Accounts Flow', checked: false },
  { name: 'standard_flow_enabled', label: 'Authorization Code Flow', checked: false },
  { name: 'implicit_flow_enabled', label: 'Implicit Flow', checked: false },
  { name: 'direct_access_grants_enabled', label: 'Direct Access Grant Flow', checked: false }
]

const OIDC_JWT_SETTINGS_DEFAULTS = {
  type: {
    value: 'plain',
    name: 'jwt_claim_with_client_id_type',
    label: 'ClientID Token Claim Type',
    hint: 'Process the ClientID Token Claim value as a string or as a liquid template. When set to "Liquid" you can define more complex rules. e.g. If "some_claim" is an array you can select the first value this like {{ some_claim | first }}.',
    catalog: {
      plain: 'plain',
      liquid: 'liquid'
    }
  },
  item: {
    value: 'azp',
    name: 'jwt_claim_with_client_id',
    label: 'ClientID Token Claim',
    placeholder: 'azp',
    hint: 'The Token Claim that contains the clientID. Defaults to "azp".'
  }
}

const OIDC_SETTINGS_DEFAULTS = {
  basicSettings: OIDC_BASICS_SETTINGS_DEFAULTS,
  flowSettings: OIDC_FLOW_SETTINGS_DEFAULTS,
  jwtSettings: OIDC_JWT_SETTINGS_DEFAULTS
}

const CREDENTIALS_LOCATION_DEFAULTS = {
  value: 'headers',
  name: 'credentials_location',
  catalog: {
    headers: 'As HTTP Headers',
    query: 'As query parameters (GET) or body parameters (POST/PUT/DELETE)',
    authorization: 'As HTTP Basic Authentication'
  }
}

const SECURITY_DEFAULTS = [
  {
    defaultValue: '',
    placeholder: 'https://api.provider-name.com',
    label: 'Host Header',
    name: 'hostname_rewrite',
    hint: 'Lets you define a custom Host request header. This is needed if your API backend only accepts traffic from a specific host.',
    value: '',
    readOnly: false
  },
  {
    defaultValue: '',
    placeholder: 'https://api.provider-name.com',
    label: 'Secret Token',
    name: 'secret_token',
    hint: 'Enables you to block any direct developer requests to your API backend; each 3scale API gateway call to your API backend contains a request header called X-3scale-proxy-secret-token. The value of this header can be set by you here. It\'s up to you ensure your backend only allows calls with this secret header.',
    value: '',
    readOnly: false
  }
]

const GATEWAY_RESPONSE_DEFAULT = [
  {
    legend: 'Authentication Failed Error',
    collection: [
      {
        label: 'Response Code',
        name: 'error_status_auth_failed',
        value: '403',
        inputType: 'number'
      },
      {
        label: 'Content-type',
        name: 'error_headers_auth_failed',
        value: 'text/plain; charset=us-ascii',
        inputType: 'text'
      },
      {
        label: 'Response Body',
        name: 'error_auth_failed',
        value: 'Authentication failed',
        inputType: 'text'
      }
    ]
  },
  {
    legend: 'Authentication Missing Error',
    collection: [
      {
        label: 'Response Code',
        name: 'error_status_auth_missing',
        value: '403',
        inputType: 'number'
      },
      {
        label: 'Content-type',
        name: 'error_headers_auth_missing',
        value: 'text/plain; charset=us-ascii',
        inputType: 'text'
      },
      {
        label: 'Response Body',
        name: 'error_auth_missing',
        value: 'Authentication parameters missing',
        inputType: 'text'
      }
    ]
  },
  {
    legend: 'Match Error',
    collection: [
      {
        label: 'Response Code',
        name: 'error_status_no_match',
        value: '404',
        inputType: 'number'
      },
      {
        label: 'Content-type',
        name: 'error_headers_no_match',
        value: 'text/plain; charset=us-ascii',
        inputType: 'text'
      },
      {
        label: 'Response Body',
        name: 'error_no_match',
        value: 'No Mapping Rule matched',
        inputType: 'text'
      }
    ]
  },
  {
    legend: 'Usage limit exceeded error',
    collection: [
      {
        label: 'Response Code',
        name: 'error_status_limits_exceeded',
        value: '429',
        inputType: 'number'
      },
      {
        label: 'Content-type',
        name: 'error_headers_limits_exceeded',
        value: 'text/plain; charset=us-ascii',
        inputType: 'text'
      },
      {
        label: 'Response Body',
        name: 'error_limits_exceeded',
        value: 'Usage limit exceeded',
        inputType: 'text'
      }
    ]
  }
]

const AUTHENTICATION_SETTINGS_DEFAULT = {
  oidcSettings: OIDC_SETTINGS_DEFAULTS,
  appIdKeyPairSettings: APP_ID_KEY_PAIR_SETTINGS_DEFAULT,
  apiKeySettings: API_KEY_SETTINGS_DEFAULT
}

const SETTINGS_DEFAULT = {
  isProxyCustomUrlActive: false,
  integrationMethod: INTEGRATION_METHOD_DEFAULTS,
  authenticationMethod: AUTHENTICATION_METHOD_DEFAULTS,
  proxyEndpoints: PROXY_ENDPOINTS_DEFAULTS,
  authenticationSettings: AUTHENTICATION_SETTINGS_DEFAULT,
  credentialsLocation: CREDENTIALS_LOCATION_DEFAULTS,
  security: SECURITY_DEFAULTS,
  gatewayResponse: GATEWAY_RESPONSE_DEFAULT
}

export {
  INTEGRATION_METHOD_DEFAULTS,
  PROXY_ENDPOINTS_DEFAULTS,
  AUTHENTICATION_METHOD_DEFAULTS,
  AUTHENTICATION_SETTINGS_DEFAULT,
  OIDC_SETTINGS_DEFAULTS,
  SETTINGS_DEFAULT
}
