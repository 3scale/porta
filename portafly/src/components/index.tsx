export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
