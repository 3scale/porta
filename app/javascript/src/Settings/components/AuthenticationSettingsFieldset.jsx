// @flow

import * as React from 'react'
import { FormFieldset, FormLegend } from 'Form'
import { FormCollection, TextInputGroup } from 'Settings/components/Common'
import { OidcFieldset } from 'Settings/components/OidcFieldset'
import type { FieldGroupProps, TypeItemProps } from 'Settings/types'

const OIDC_AUTH_METHOD = 'oidc'
const API_KEY_METHOD = '1'
const APP_ID_KEY_METHOD = '2'

type Props = {
  isServiceMesh: boolean,
  authenticationMethod: string,
  apiKeySettings: FieldGroupProps,
  appIdKeyPairSettings: FieldGroupProps[],
  oidcSettings: {
    basicSettings: TypeItemProps,
    flowSettings: FieldGroupProps[],
    jwtSettings: TypeItemProps
  }
}

const AuthenticationSettingsFieldset = ({
  isServiceMesh,
  authenticationMethod,
  apiKeySettings,
  appIdKeyPairSettings,
  oidcSettings
}: Props): React.Node => {
  const isOidc = authenticationMethod === OIDC_AUTH_METHOD
  const isApiKey = authenticationMethod === API_KEY_METHOD
  const isAppIdKey = authenticationMethod === APP_ID_KEY_METHOD
  return (
    (!isServiceMesh || isOidc) && <FormFieldset id='fieldset-AuthenticationSettings'>
      <FormLegend>Authentication Settings</FormLegend>
      { isApiKey && <FormCollection collection={[apiKeySettings]} ItemComponent={TextInputGroup} legend='API KEY (USER_KEY) BASICS' /> }
      { isAppIdKey && <FormCollection collection={appIdKeyPairSettings} ItemComponent={TextInputGroup} legend='APP_ID AND APP_KEY PAIR BASICS' /> }
      { isOidc && <OidcFieldset {...oidcSettings} isServiceMesh={isServiceMesh} /> }
    </FormFieldset>
  )
}

export {
  AuthenticationSettingsFieldset
}
