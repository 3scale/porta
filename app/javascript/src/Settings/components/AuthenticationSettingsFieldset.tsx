import * as React from 'react'

import { FormCollection, FormFieldset, FormLegend, TextInputGroup } from 'Settings/components/Common'
import { OidcFieldset } from 'Settings/components/OidcFieldset'
import { FieldGroupProps, TypeItemProps } from 'Settings/types'

const OIDC_AUTH_METHOD = 'oidc'
const API_KEY_METHOD = '1'
const APP_ID_KEY_METHOD = '2'

type Props = {
  isServiceMesh: boolean,
  authenticationMethod: string, // TODO: use exact types
  apiKeySettings: FieldGroupProps,
  appIdKeyPairSettings: FieldGroupProps[],
  oidcSettings: {
    basicSettings: TypeItemProps,
    flowSettings: FieldGroupProps[],
    jwtSettings: TypeItemProps
  }
};

const AuthenticationSettingsFieldset: React.FunctionComponent<Props> = ({
  isServiceMesh,
  authenticationMethod,
  apiKeySettings,
  appIdKeyPairSettings,
  oidcSettings
}) => {
  const isOidc = authenticationMethod === OIDC_AUTH_METHOD
  const isApiKey = authenticationMethod === API_KEY_METHOD
  const isAppIdKey = authenticationMethod === APP_ID_KEY_METHOD
  return (
    (!isServiceMesh || isOidc) && (
      <FormFieldset id='fieldset-AuthenticationSettings'>
        <FormLegend>Authentication Settings</FormLegend>
        { isApiKey && <FormCollection collection={[apiKeySettings]} ItemComponent={TextInputGroup} legend='API KEY (USER_KEY) BASICS' /> }
        { isAppIdKey && <FormCollection collection={appIdKeyPairSettings} ItemComponent={TextInputGroup} legend='APP_ID AND APP_KEY PAIR BASICS' /> }
        { isOidc && <OidcFieldset {...oidcSettings} isServiceMesh={isServiceMesh} /> }
      </FormFieldset>
    )
  ) as React.ReactElement // Hack: Shortcircuit is not supported yet. See: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18912
}

export { AuthenticationSettingsFieldset, Props }
