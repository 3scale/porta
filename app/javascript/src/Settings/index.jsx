// @flow

import * as React from 'react'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { Form } from 'Settings/components/Form'
import { SETTINGS_DEFAULT } from 'Settings/defaults'
import type { SettingsProps } from 'Settings/types'

type Props = {
  settings: SettingsProps,
  elementId: string
}

const initSettings = ({
  settings = SETTINGS_DEFAULT,
  elementId
}: Props): void => createReactWrapper(<Form {...settings} />, elementId)

export {
  initSettings
}
