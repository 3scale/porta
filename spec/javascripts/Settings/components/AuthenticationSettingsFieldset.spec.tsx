import React from 'react'
import { shallow } from 'enzyme'
import { AuthenticationSettingsFieldset } from 'Settings/components/AuthenticationSettingsFieldset'
import { AUTHENTICATION_SETTINGS_DEFAULTS } from 'Settings/defaults'

function setup (customProps = {}) {
  const props = {
    ...AUTHENTICATION_SETTINGS_DEFAULTS,
    ...{
      isServiceMesh: false,
      authenticationMethod: '1'
    },
    ...customProps
  }

  const view = shallow(<AuthenticationSettingsFieldset {...props} />)

  return { view, props }
}

it('should render correctly', () => {
  const { view } = setup()
  expect(view).toMatchSnapshot()
})

it('should not render when Service Mesh is active', () => {
  const customProps = { isServiceMesh: true }
  const { view } = setup(customProps)
  expect(view).toMatchSnapshot()
})

it('should render only OIDC when Service Mesh is active and Oidc method is selected', () => {
  const customProps = { isServiceMesh: true, authenticationMethod: 'oidc' }
  const { view } = setup(customProps)
  expect(view).toMatchSnapshot()
})
