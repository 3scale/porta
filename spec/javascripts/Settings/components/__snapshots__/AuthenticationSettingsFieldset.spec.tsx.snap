// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render when Service Mesh is active 1`] = `""`;

exports[`should render correctly 1`] = `
<FormFieldset
  id="fieldset-AuthenticationSettings"
>
  <FormLegend>
    Authentication Settings
  </FormLegend>
  <FormCollection
    ItemComponent={[Function]}
    collection={
      Array [
        Object {
          "label": "Auth user key",
          "name": "auth_user_key",
          "value": "user_key",
        },
      ]
    }
    legend="API KEY (USER_KEY) BASICS"
  />
</FormFieldset>
`;

exports[`should render only OIDC when Service Mesh is active and Oidc method is selected 1`] = `
<FormFieldset
  id="fieldset-AuthenticationSettings"
>
  <FormLegend>
    Authentication Settings
  </FormLegend>
  <OidcFieldset
    basicSettings={
      Object {
        "item": Object {
          "hint": "Location of your OpenID Provider. The format of this endpoint is determined on your OpenID Provider setup. A common guidance would be \\"https://CLIENT_ID:CLIENT_SECRET@HOST:PORT/auth/realms/REALM_NAME\\".",
          "label": "OpenID Connect Issuer",
          "name": "oidc_issuer_endpoint",
          "placeholder": "https://sso.example.com/auth/realms/gateway",
          "value": "",
        },
        "type": Object {
          "catalog": Object {
            "keycloak": "Red Hat Single Sign-On",
            "rest": "REST API",
          },
          "label": "OpenID Connect Issuer Type",
          "name": "oidc_issuer_type",
          "value": "keycloak",
        },
      }
    }
    flowSettings={
      Array [
        Object {
          "checked": false,
          "label": "Service Accounts Flow",
          "name": "service_accounts_enabled",
        },
        Object {
          "checked": false,
          "label": "Authorization Code Flow",
          "name": "standard_flow_enabled",
        },
        Object {
          "checked": false,
          "label": "Implicit Flow",
          "name": "implicit_flow_enabled",
        },
        Object {
          "checked": false,
          "label": "Direct Access Grant Flow",
          "name": "direct_access_grants_enabled",
        },
      ]
    }
    isServiceMesh={true}
    jwtSettings={
      Object {
        "item": Object {
          "hint": "The Token Claim that contains the clientID. Defaults to \\"azp\\".",
          "label": "ClientID Token Claim",
          "name": "jwt_claim_with_client_id",
          "placeholder": "azp",
          "value": "azp",
        },
        "type": Object {
          "catalog": Object {
            "liquid": "liquid",
            "plain": "plain",
          },
          "hint": "Process the ClientID Token Claim value as a string or as a liquid template. When set to \\"Liquid\\" you can define more complex rules. e.g. If \\"some_claim\\" is an array you can select the first value this like {{ some_claim | first }}.",
          "label": "ClientID Token Claim Type",
          "name": "jwt_claim_with_client_id_type",
          "value": "plain",
        },
      }
    }
  />
</FormFieldset>
`;
